Source/JavaScriptCore/wasm/js/WasmToJS.cpp

```diff
@@ -492,6 +510,10 @@ void emitThrowWasmToJSException(CCallHelpers& jit, GPRReg wasmInstance, Wasm::Ex
     jit.loadPtr(CCallHelpers::Address(GPRInfo::argumentGPR0), GPRInfo::argumentGPR0);
     jit.copyCalleeSavesToEntryFrameCalleeSavesBuffer(GPRInfo::argumentGPR0);
     jit.move(GPRInfo::callFrameRegister, GPRInfo::argumentGPR0);
+
+    if (wasmInstance != GPRInfo::argumentGPR2)
```

Comment:
Isn't this pinned? I am confused now

Reply:
Hmm, what is the difference between this and wasmContextInstanceGPR?

Reply:
Since the operation that's being called (`operationWasmToJSException`) takes the instance as the third argument, the instance pointer has to be moved into GPR2 or be there already.

Previously all callers of this function just passed in GPR2 (and it would fail in weird ways if you passed in other registers), so I just changed it here to work if you passed a different one in (such as the pinned register).